// components/search/index.js
import React, { Component } from 'react'
import { connect } from 'react-redux'

import { searchHouse } from 'actions/index'

class Search extends Component {



    state = {
        search: {
            product: '',
            garage: '',
            type: ''
        }
    }

    componentDidMount() {
        this.setState({
            search: {
                product: this.state.search.product,
                garage: this.state.search.garage,
                type: this.state.search.type
            }
        });

    }

    reset = (event) => {
        event.preventDefault();
        this.setState({
            search: {
                product: '',
                garage: '',
                type: ''
            }
        });


    }

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.searchHouse(this.state.search)

    }

    handleChangeFor = (propertyName) => (event) => {
        const { search } = this.state;
        const newContact = {
            ...search,
            [propertyName]: event.target.value
        };
        this.setState({ search: newContact });
    }


    render() {
        return (
            <div >
                <h3 className='lead'>Quick search</h3>
                <div className='input-group d-flex'>
                    {/* <form >
                        <select class="form-select mb-1" aria-label="Default select example">
                            <option selected>Sort price by</option>
                            <option value="1" onClick={alert('Increase')}>Increase</option>
                            <option value="2" onClick={alert('Decrease')}>Decrease</option>
                            
                        </select>
                    </form> */}

                    <form onSubmit={this.handleSubmit}>
                        <input className='form-control mb-1' type="text" placeholder="Search product here..." onChange={this.handleChangeFor('product')} value={this.state.search.product} />
                        <input className='form-control mb-1' type="text" placeholder="Search garage here..." onChange={this.handleChangeFor('garage')} value={this.state.search.garage} />
                        <input className='form-control mb-1' type="text" placeholder="Search type here..." onChange={this.handleChangeFor('type')} value={this.state.search.type} />
                        <span className='input-group-btn' >

                            <button className="btn btn-primary" onClick={this.handleSubmit} >Search  <i className="fa fa-search" aria-hidden="true" ></i></button>
                            <button className="btn btn-primary pull-right" onClick={this.reset}>Reset</button>
                        </span>

                    </form>

                </div>

            </div>
        )
    }
}

const mapDispatchToProps = {
    searchHouse
}
export default connect(null, mapDispatchToProps)(Search)
